terraform/modules/kubernetes/monitoring.tf

```diff
@@ -115,3 +115,35 @@ resource "azurerm_monitor_metric_alert" "redis_memory" {
     ]
   }
 }
+data "azurerm_monitor_diagnostic_categories" "postgres" {
+  count = var.deploy_azure_backing_services ? 1 : 0
+
+  resource_id = azurerm_postgresql_flexible_server.postgres-server[0].id
+}
+
+resource "azurerm_log_analytics_workspace" "postgres" {
+  count = var.deploy_azure_backing_services ? 1 : 0
+
+  name                = "${azurerm_postgresql_flexible_server.postgres-server[0].name}-logs"
+  location            = data.azurerm_resource_group.app-resource-group.location
+  resource_group_name = data.azurerm_resource_group.app-resource-group.name
+  sku                 = "PerGB2018"
+}
+resource "azurerm_monitor_diagnostic_setting" "postgres" {
+  count = var.deploy_azure_backing_services ? 1 : 0
+
+  name                       = "${azurerm_postgresql_flexible_server.postgres-server[0].name}-diag"
+  target_resource_id         = azurerm_postgresql_flexible_server.postgres-server[0].id
+  log_analytics_workspace_id = azurerm_log_analytics_workspace.postgres[0].id
+
+  dynamic "enabled_log" {
+    for_each = data.azurerm_monitor_diagnostic_categories.postgres[0].log_category_types
+    content {
+      category = enabled_log.value
+
+      retention_policy {
```

Comment:
The doc says this block is optional: https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/monitor_diagnostic_setting#retention_policy
Can you try to remove it?